lib/iser.c

```diff
@@ -812,6 +816,32 @@ static int iser_create_iser_conn_res(struct iser_conn *iser_conn) {
 	return ret;
 }
 
+/*
+ * iser_rdma_set_option() - set rdma options
+ *
+ * @cma_id:    connection manager id
+ * @iscsi:   iscsi context
+ */
+int iser_rdma_set_option(__attribute__((unused)) struct rdma_cm_id *cma_id, __attribute__((unused)) struct iscsi_context *iscsi) {
+#ifdef HAVE_RDMA_ACK_TIMEOUT
+	int ret = 0;
+	unsigned char timeout = 8;
+
+	/* calculate according to the formula 4.096 * 2^(timeout) usec.
+	   8 means 1048.576 usec (0.00104 sec) */
+	ret = rdma_set_option(cma_id, RDMA_OPTION_ID, RDMA_OPTION_ID_ACK_TIMEOUT, &timeout, sizeof(timeout));
```

Comment:
It seems weird to me to set the ACK timeout to a fixed value. Shouldn't the ACK timeout depend on the subnet timeout? See e.g. https://lore.kernel.org/linux-rdma/20171011172729.16984-9-bart.vanassche@wdc.com/

Reply:
If a user wants to use a specified value, how about passing it by getenv? Keep the logic as similar as tcp keepalive setting.
A example may like this:
if (value = getevn("LIBISCSI_RDMA_ACK_TIMEOUT")) {
    timeout = atoi(value);
    rdma_set_option....
}